pyls/server.py

```diff
@@ -32,6 +43,9 @@ def handle(self):
                 log.exception("Language server shutting down for uncaught exception")
                 break
 
+            if isinstance(self.wfile, socket_type):
+                done = True
```

Comment:
I think whoever wrote this code was more than a bit confused as to how `socketserver.ThreadingTCPServer` works, and in particular the role of the handler object:

```
The request handler class must define a new handle() method, and can override any of the following methods. A new instance is created for each request.
```